dbms/src/Client/Connection.cpp

```diff
@@ -455,7 +456,10 @@ void Connection::sendData(const Block & block, const String & name)
         block_out = std::make_shared<NativeBlockOutputStream>(*maybe_compressed_out, server_revision, block.cloneEmpty());
     }
 
-    writeVarUInt(Protocol::Client::Data, *out);
+    if (scalar)
```

Comment:
Do we really need backward incompatible changes here?

Reply:
It's compatible. Old servers just don't send any scalars

Reply:
Hmm, what kind of backward incompatibility do you refer? Do you mean running a cluser with both old and new versions of clickhouse instances?

Reply:
Yes. It looks like we cannot upgrage or downgrade without downtime. Some installations has different update time for nodes with Distributed tables and their data sources. It's better to keep possibility to upgrade or downgrade them separately.

Reply:
ok, I can add an scalar optimization option and default to false if that helps